src/Directory.Build.props

```diff
@@ -90,9 +94,13 @@
   </PropertyGroup>
 
   <!-- Ensure ns2.0 ref assemblies are placed under `ref/netstandard2.0` in the NuGet package -->
-  <Target Name="ShipRefAssembliesToNuGetPackage" Condition="'$(TargetFramework)' == 'netstandard2.0'">
+  <Target Name="ShipRefAssembliesToNuGetPackage" BeforeTargets="Pack" Condition="$(IsInnerBuild) == true">
     <ItemGroup>
-      <TfmSpecificPackageFile Include="@(BuiltProjectOutputGroupOutput);@(FinalDocFile)">
+      <TfmSpecificPackageFile Include="@(IntermediateRefAssembly);@(FinalDocFile)">
```

Comment:
So I'd say you should use `$(TargetRefPath)` here; it should be identical for all realistic official build cases but that's the "final" ref assembly location; `@(IntermediateRefAssembly)` is written to by the compiler every time.